src/main/java/org/folio/services/reindex/ReindexJobRunner.java

```diff
@@ -47,26 +51,40 @@ public ReindexJobRunner(Context vertxContext, Map<String, String> okapiHeaders)
       vertxContext,
       new CommonDomainEventPublisher<>(vertxContext, okapiHeaders,
         KafkaTopic.instance(tenantId(okapiHeaders), environmentName())),
+      new CommonDomainEventPublisher<>(vertxContext, okapiHeaders,
+        KafkaTopic.authority(tenantId(okapiHeaders), environmentName())),
       tenantId(okapiHeaders));
   }
 
   public ReindexJobRunner(PostgresClientFuturized postgresClient, ReindexJobRepository repository,
-    Context vertxContext, CommonDomainEventPublisher<Instance> domainEventPublisher,
-    String tenantId) {
+                          Context vertxContext, CommonDomainEventPublisher<Instance> domainEventPublisher,
+                          CommonDomainEventPublisher<Authority> authorityEventPublisher,
+                          String tenantId) {
 
     this.postgresClient = postgresClient;
     this.reindexJobRepository = repository;
-    this.eventPublisher = domainEventPublisher;
+    this.instanceEventPublisher = domainEventPublisher;
+    this.authorityEventPublisher = authorityEventPublisher;
     this.tenantId = tenantId;
 
     initWorker(vertxContext);
   }
 
-  public void startReindex(ReindexJob reindexJob) {
+  public void startReindex(ReindexJob reindexJob, ReindexResourceName reindexResourceName) {
     workerExecutor.executeBlocking(
-      promise -> streamInstanceIds(new ReindexContext(reindexJob))
-        .map(notUsed -> null)
-        .onComplete(promise))
+        promise -> {
+          switch (reindexResourceName) {
```

Comment:
switch without "default" and without last "break" looks weird especially in lambda.